synapse/event_auth.py

```diff
@@ -168,13 +168,23 @@ async def check_state_independent_auth_rules(
         return
 
     # 2. Reject if event has auth_events that: ...
-    auth_events = await store.get_events(
-        event.auth_event_ids(),
-        redact_behaviour=EventRedactBehaviour.as_is,
-        allow_rejected=True,
-    )
     if batched_auth_events:
-        auth_events.update(batched_auth_events)
+        auth_event_ids = event.auth_event_ids()
+        auth_events = dict(batched_auth_events)
+        if set(auth_event_ids) - batched_auth_events.keys():
+            auth_events.update(
+                await store.get_events(
+                    set(auth_event_ids) - batched_auth_events.keys(),
+                    redact_behaviour=EventRedactBehaviour.as_is,
+                    allow_rejected=True,
+                )
+            )
```

Comment:
It might make more sense to keep the difference instead of calculating it twice:

```suggestion
        auth_event_ids = event.auth_event_ids()
        auth_events = dict(batched_auth_events)
        needed_auth_event_ids = set(event.auth_event_ids()) - batched_auth_events.keys()
        if needed_auth_event_ids:
            auth_events.update(
                await store.get_events(
                    needed_auth_event_ids,
                    redact_behaviour=EventRedactBehaviour.as_is,
                    allow_rejected=True,
                )
            )
```

Reply:
thanks, that makes sense!